lib/nodes/iec61850_goose.cpp

```diff
@@ -0,0 +1,873 @@
+			break;
+		case MmsType::MMS_FLOAT:
+			if (float_size != -1)
+				meta = {.size = float_size };
+			break;
+		default:
+			break;
+		}
+
+		auto signal_data = SignalData {};
+
+		if (value_json) {
+			ret = signal_data.parseJson(goose_type->signal_type, value_json);
+			if (ret) return ret;
+		}
+
+		auto signal = std::optional<int> {};
+
+		if (signal_str)
+			signal = out.signals->getIndexByName(signal_str);
+
+		OutputData value = {
+			.signal = signal,
+			.default_value = GooseSignal { goose_type, signal_data, meta }
+		};
+
+		data.push_back(value);
+	};
+
+	return 0;
+}
+
+int GooseNode::parsePublisher(json_t *json, json_error_t *err, PublisherConfig &pc)
+{
+	int ret;
+
+	char *go_id = nullptr;
+	char *go_cb_ref = nullptr;
+	char *data_set_ref = nullptr;
+	char *dst_address_str = nullptr;
+	int app_id = 0;
+	int conf_rev = 0;
+	int time_allowed_to_live = 0;
+	json_t *data_json = nullptr;
+	ret = json_unpack_ex(json, err, 0, "{ s:s, s:s, s:s, s:s, s:i, s:i, s:i, s:o }",
+		"go_id", &go_id,
+		"go_cb_ref", &go_cb_ref,
+		"data_set_ref", &data_set_ref,
+		"dst_address", &dst_address_str,
+		"app_id", &app_id,
+		"conf_rev", &conf_rev,
+		"time_allowed_to_live", &time_allowed_to_live,
+		"data", &data_json
+	);
+	if (ret) return ret;
+
+	std::optional dst_address = stringToMac(dst_address_str);
+	if (!dst_address)
+		throw RuntimeError("Invalid dst_address");
+
+	pc.go_id = std::string { go_id };
+	pc.go_cb_ref = std::string { go_cb_ref };
+	pc.data_set_ref = std::string { data_set_ref };
+	pc.dst_address = *dst_address;
+	pc.app_id = app_id;
+	pc.conf_rev = conf_rev;
+	pc.time_allowed_to_live = time_allowed_to_live;
+
+	parsePublisherData(data_json, err, pc.data);
+	if (ret) return ret;
+
+	return 0;
+}
+
+int GooseNode::parsePublishers(json_t *json, json_error_t *err, std::vector<OutputContext> &ctx)
+{
+	int ret;
+	int index;
+	json_t* publisher_json;
+
+	json_array_foreach(json, index, publisher_json) {
+		PublisherConfig pc;
+
+		ret = parsePublisher(publisher_json, err, pc);
+		if (ret) return ret;
+
+		ctx.push_back(OutputContext { pc });
+	}
+
+	return 0;
+}
+
+std::vector<int> GooseNode::getPollFDs()
+{
+	return { queue_signalled_fd(&input.queue) };
+}
+
+int GooseNode::prepare()
+{
+	int ret;
+
+	ret = pool_init(&input.pool, input.queue_length, SAMPLE_LENGTH(getInputSignals(false)->size()));
+	if (ret)
+		return ret;
+
+	ret = queue_signalled_init(&input.queue, input.queue_length);
+	if (ret)
+		return ret;
+
+	return Node::prepare();
+}
+
+int GooseNode::start()
+{
+	if (in.enabled)
+		startReceiver();
+
+	if (out.enabled)
+		startPublishers();
+
+	return Node::start();
+}
+
+int GooseNode::stop()
+{
+	int err __attribute__((unused));
+
+	stopReceiver();
+	stopPublishers();
+
+	err = queue_signalled_close(&input.queue);
+
+	return Node::stop();
+}
+
+static char name[] = "iec61850-goose";
+static char description[] = "Subscribe to and publish GOOSE messages";
+static NodePlugin<
+	GooseNode,
```

Comment:
Hi while I like this way of formatting the template args, its not consistent with the rest of the code.

Would you mind to use the same style as in the other files?

I would be happy to accept a dedicated PR which changes the formatting to your new style project-wide.